dll/win32/shell32/CDefView.cpp

```diff
@@ -587,28 +591,54 @@ void CDefView::UpdateStatusbar()
             }
         }
 
+        szPartText[0] = szPartText2[0] = UNICODE_NULL;
+
         /* Don't show the file size text if there is 0 bytes in the folder
          * OR we only have folders selected. */
         if ((cSelectedItems && !bIsOnlyFoldersSelected) || uTotalFileSize)
-        {
             StrFormatByteSizeW(uTotalFileSize, szPartText, _countof(szPartText));
-        }
-        else
-        {
-            *szPartText = 0;
-        }
 
-        m_pShellBrowser->SendControlMsg(FCW_STATUS, SB_SETTEXT, 1, (LPARAM)szPartText, &lResult);
+        if (hThread != m_hUpdateStatusbarThread)
+            return;
 
         /* If we are in a Recycle Bin folder then show no text for the location part. */
         if (!_ILIsBitBucket(m_pidlParent))
         {
-            LoadStringW(shell32_hInstance, IDS_MYCOMPUTER, szPartText, _countof(szPartText));
+            LoadStringW(shell32_hInstance, IDS_MYCOMPUTER, szPartText2, _countof(szPartText2));
             pIcon = (LPARAM)m_hMyComputerIcon;
         }
 
+        if (hThread != m_hUpdateStatusbarThread)
```

Comment:
Not sure it's useful to have that check just after loading the string, while the thread check has been done just above.

Reply:
Deleted.